### DISCUSSION TITLE
Zen folders like in arc

### DISCUSSION THREAD

**Discussion by gal1le0-cl** [zen-browser/desktop#3152] [Ideas]
If you add this, i think lot of ARC users are going to pass into Zen

**Comment by yumin-chen** [comment]
I second this. This has been a feature I can't live without. I can use it to apply PARA method within the workspace and it is incredibly useful.

Example from Arc:
<img width="490" alt="Screenshot 2024-12-11 at 7 42 55 PM" src="https://github.com/user-attachments/assets/a4d06df7-8d4d-4d50-bec5-5e54a132a36c" />

https://github.com/user-attachments/assets/b987d2db-c996-4732-8fe9-3098891386e8

**Reply by pranavgoel29** [reply]
I second this

**Reply by soundstep** [reply]
same

**Reply by AlexKrew** [reply]
same

**Reply by lazylagom** [reply]
same here!

**Reply by Kaspazza** [reply]
Wanted to try zen, but without directories it's a no-go for me

**Reply by acexiis** [reply]
Same here, even if Arc is dead, I cannot move to Zen without the folder feature.

**Reply by gabrielcachioniclutch** [reply]
+1

**Reply by electroheadfx** [reply]
me too

**Reply by Melvynx** [reply]
agree

**Comment by RayZ3R0** [comment]
We are letting firefox finish their implementation of tab groups so we can build upon it, hence it will probably take a while for Zen to have something like that

**Reply by smiling-fighting** [reply]
how long does it seem like it'll be until then?

**Reply by syscat29** [reply]
> how long does it seem like it'll be until then?

Tab groups are out of experimental in Firefox 135 which is slated for Feb 4.

**Reply by james-arawhanui** [reply]
Great, at least it sounds like they have momentum

**Reply by moodjx** [reply]
Tried it on firefox 135 -> browser.tabs.groups.enabled true
Looks better. Can't wait to see this update on Zen

**Reply by tobias-hay** [reply]
> Tried it on firefox 135 -> browser.tabs.groups.enabled true
> 
> Looks better. Can't wait to see this update on Zen

Not ideal, they don't allow nesting iirc.

**Reply by kanokanor** [reply]
Have you considered to use this approach for tab grouping instead to support nesting?
https://github.com/zen-browser/desktop/discussions/3152#discussioncomment-11953691

**Reply by tobias-hay** [reply]
> Have you considered to use this approach for tab grouping instead to support nesting?
> 
> https://github.com/zen-browser/desktop/discussions/3152#discussioncomment-11953691

I think he mentioned that would require some rewriting & reworking because you can't open bookmarks inline by default. I would look into it but ran into issues when trying to build the dev env lol.

**Reply by andrezzasouza** [reply]
Hello, @RayZ3R0! Zen is awesome! However, as other users mentioned, that's the only thing I need to move from Arc to Zen. Do you have a timeline for its release? I'm anxiously waiting here! 🙃

**Reply by DamienBlackwood** [reply]
Okay thank you

**Comment by yumin-chen** [comment]
Related: https://github.com/zen-browser/desktop/discussions/2618

**Comment by ldinino** [comment]
Just wanted to mention that Mozilla has just begun implementing tab groups into their Nightly builds:

https://bugzilla.mozilla.org/show_bug.cgi?id=1907090 

https://bugzilla.mozilla.org/show_bug.cgi?id=1938187

https://hg.mozilla.org/mozilla-central/rev/a1ddc48882ff

I did some tests and it seems to work pretty well! 

![Screenshot 2024-12-30 at 1 23 26 PM](https://github.com/user-attachments/assets/dea1333b-6ca5-4db2-998e-c7c5cbfeb555)


This would really round out the experience in Zen.

**Reply by sinni800** [reply]
They just can't be moved currently. Is there any hack to that that working? I wonder if we could style them as well using a zen mod.

**Comment by BraveBrowserFan** [comment]
Hi

On Brave Beta browser has folders for group tabs. I use this addon Simple Tab Groups at https://addons.mozilla.org/en-US/firefox/addon/simple-tab-groups/ Also it can backup the bookmarks in the settings. You create a  new group for each spaces you have. Then you click the group that you open. leave it open when you shut down the computer then when you go back on zen it will have the same open group.  

![image](https://github.com/user-attachments/assets/e8ae027e-04f7-40c5-b661-8f50f263d0b4)

**Comment by BraveBrowserFan** [comment]
Hi

On Brave browser they have 
![image](https://github.com/user-attachments/assets/7922d610-f03a-44c0-b367-96c6cea785c2)
then i make group tabs for each space. I right click and use 
![image](https://github.com/user-attachments/assets/5e077a9c-8a49-408e-9fc3-28c624076a4e)

**Comment by IAmTahazzot** [comment]
## Important

1. Just to mention that, each `container/ workspace` should have it's own folders/ dir's to work with. Similar to Arc. It seems more reasonable than having same dir, folder for all `container/ workspace`
2. It would be nice if those dir or folder are scroll-able if there's too many tabs opened. The whole tab bar from dir/ folder to all tabs that are opened. same as Arc browser

**Comment by PaulMcF1987** [comment]
@RayZ3R0 @ldinino 
But... I used to use sidekick who offer Chromes Tab goups and then I found Arc which takes it one step further and uses folders and also allows sub folders/nested folders

One particular area where I prefer the use of folders is. In sikdekick, whenever I open the browser, the tab groups reopen and I need to go through and close them. In arc, the folders are in the same state as they were before I closed the browser.
Also, for organising, the use of sub folders is a game changer

**Comment by tobias-hay** [comment]
Jumping onto this thread to say i also would like this. Hopefully mozzilla can implement their stable version soon so we can get it in zen

**Comment by 12th-devs** [comment]
https://github.com/Anoms12/Advanced-Tab-Groups

**Reply by yumin-chen** [reply]
Nice!!!

**Reply by jazzroutine** [reply]
Super! Thank you for that find!

**Comment by PaulMcF1987** [comment]
@RayZ3R0 @ldinino 
Going back to my last post above and based on the pictures that others have shared on what they like about the set up on arc...

I personally feel that it would be better to base thus feature type off of sideberry rather than mozillas tab grouping

The reason being, as mentioned above. I have used sidekick who base it on Chromes grouping option which is very limited.
Arc use a set up which is similar to sideberry which offers nested folders and it offers a lot more flexibility.

A solution based on Sideberry would definitely be an excellent addition

**Comment by issuex** [comment]
to dev, why not just add a special folder to bookmark and use it for store tab data that user can also rename tab or add folder or sync it, make ui to combine tab and the bookmark that store in special folder to make it look like using normal tab.

for example.
/ Bookmark Menu / ZEN-TAB / WORKSPACE-1 / <all tab here, store as bookmark >
/ Bookmark Menu / ZEN-TAB / WORKSPACE-2 /< user dir >/ <all tab here, store as bookmark >

**Reply by yumin-chen** [reply]
Good idea! Syncs well automatically too

**Reply by JSGRD22** [reply]
I already use bookmarks to back up and sync tabs manually, this is a good idea

**Reply by cornelius-99** [reply]
I really would prefer this approach because, with the firefox tab-groups it's not possible to have nested folders like in arc

**Reply by pressRtowin** [reply]
As described above, it seems that this wouldn't preserve information about what container a tab belongs in.

**Comment by kanokanor** [comment]
Another great feature would be that able to open child tabs like in Sidebery or SigmaOS

**Comment by DamienBlackwood** [comment]
So whats going on with this?

**Reply by rihards-simanovics** [reply]
From what I've read, the Zen team is still waiting for a stable implementation from Firefox before doing any work here: https://github.com/zen-browser/desktop/discussions/3152#discussioncomment-11537129.

**Reply by kanokanor** [reply]
I hope they consider this approach instead because Firefox tab groups does not support nesting
https://github.com/zen-browser/desktop/discussions/3152#discussioncomment-11953691

**Comment by Chaoscontrol** [comment]
Enabling tab groups in advanced config does not seem to work great so far. Also it doesn't work for pinned tabs. It caused a few bugs for me, tabs not opening properly, and also the groups not collapsing/expanding either.

**Comment by The-Uncoders** [comment]
Why not just add an optional bookmarks section in the side bar under the pinned tabs? Could be linked to the firefox bookmarks toolbar folder. That way we can create and delete bookmark folders and to return to our saved bookmarks easily and in the Native Sidebar UI?

**Comment by flight505** [comment]
I truly love the project, but I wish it were built on something other than 🔥🦊. I miss folders and WebKit. Yes, I know the team loves PCs.

**Comment by waLLxAck** [comment]
Bump. I was hyped to try it out but without folders it's a no-no. Really crucial feature.

**Comment by ldinino** [comment]
I gave the feature a try on the latest build. Going into about:config and setting **browser.tabs.groups.enabled** to **true** seems to work somewhat. It's a little jank, but works much better than it did before. Your mileage may vary.

(Tested on 1.9.1b with macOS and Windows)

**Reply by rihards-simanovics** [reply]
yeah, on Windows, you can combine the tabs into a group and name the group, but I'm assuming you can't drag the group because it is meant to slide horizontally.

**Reply by ExistentialAlex** [reply]
I found that this actually works surprisingly well. It's great for just grouping stuff together but still needs that 'Zen' flourish to really make it a part of the browser.

**Comment by PaulMcF1987** [comment]
Ill be honest, I am fearing the worst in terms of the potential folders that are on offer

Key differences between what I am used to from Arc and what it looks like is available for Zen

1 - Arc only offers folders for pinned tabs - So far Zen can only group non-pinned tab as per firefox
2 - Arc doesnt seem to rely on Chromiums grouping option and has created a stand alone folder function which when selected, adds a new folder into the pinned section which I can then drag tabs into and also nest other folders into

![image](https://github.com/user-attachments/assets/0f6e0efa-1924-4536-884f-9fd2b82a74ef)

I feel that if this is relying solely on Firefoxs grouping function, that it wont work and give the results that would make people want to switch over from Arc

I am using both at the minute, purely in the hope that Zen can offer something comparable but I think there needs to be some outside the box thinking for it to work

**Reply by PaulMcF1987** [reply]
Too add to the above...
If zen kept the grouping function for unpinned tabs and also added the folder and nested folder option for pinned thabs, I think that would give zen the edge as it would help to neaten up the pinned and unpinned section.
I have a lot of tabs I want to pin, but I have only pinned a few to keep it neat

On another note, I think Arc looks more polished in terms of its appearance especially with the pastel colours it uses, but that is a minor thing
Zen is its own browser and to compare the 2 is wrong in a sense, but there are definitely areas that I feel zen can take from Arc and even perfect

**Comment by Naatan** [comment]
Another vote for proper pinned tab folders support. I could personally not care less about folders outside of the pinned tabs. 

I've been using Arc and I love its UI/UX, the only reason I even found Zen is learning that Arc is basically being abandoned in the medium term. Frankly Zen looks awesome, and that it's Firefox based instead of chrome is a big bonus! But continuing that frankness; without the ability to group pinned tabs into collapsible folders it's just not good enough. I think this is probably the one feature I like most about Arc. I've never used bookmarks as part of my workflows, but Arc changed that with their pinned tabs feature. 

Please consider revising the priority on this feature.

**Comment by kauelima** [comment]
As chrome finally drops ManifestV2 making the full uBlock Origin not work anymore. I came to zen to test it. There are a lot of features I like, specially the firefox engine. But tab folders is a must to make the change.

**Reply by 12th-devs** [reply]
Coming next update

**Comment by Dwarph** [comment]
just came here to bump this! I was all ready to switch from arc to zen today, but I absolutely cannot without folders. I'm way too reliant on them :(

**Reply by 12th-devs** [reply]
Coming next update

https://github.com/Anoms12/Advanced-Tab-Groups

**Reply by Naatan** [reply]
This looks great! But if it's only CSS then it won't fix the issue that Firefox's tab groups don't work for pinned tabs, which in my mind is the primary use-case.

**Comment by Vulkanodox** [comment]
I don't want to be rude but Firefox already has vertical tabs and when Firefox implements the tab groups/folder there is no reason to use Zen anymore.

**Reply by mr-cheffy** [reply]
I can't watch tiktok in 4 different tabs at the same time on firefox. So that's a pass for me

**Reply by ldinino** [reply]
Vanilla Firefox makes irritating compromises and design choices that Zen fixes. There's still plenty of value beyond vertical tabs and tab groups.

**Reply by Vulkanodox** [reply]
> I can't watch tiktok in 4 different tabs at the same time on firefox. So that's a pass for me

There is an official Firefox addon for split view and multiple addons that can do it. 
I can't think of a setting where one actually would look for a 4 way split and still that can be done by the default window manager in Windows, on Linux, and even easier with a tiling window manager.

When looking at reviews and opinions about Zen people go for it because it is "that Firefox based browser with vertical tab management" 

Firefox seems to outpace the purpose of Zen for the most users. 

People have rightly suggested a more in-depth tab management with folders and nesting. But nothing was implemented.

Now Firefox itself has the features making Zen irrelevant to the majority of people who have come to Zen for this feature. 

It was by far the most upvoted and appeared in multiple discussions in the top list.

And that are just the users who bother to login to GitHub to leave a comment or idea.

I would take a guess and say that 90% of people use Zen because of the vertical tab management. Saying that other features matter more or with equal weight is not true.

**Comment by 74796a** [comment]
So having just wasted an hour of my life let me see if I've got this straight? Someone working on Zen decided it would be a good idea to arbitrarily force ALL of their users to conform to a random aesthetic preference (arguably an objective regression from a UX design perspective which i have a degree in) by breaking part of the interface without removing it? You just randomly changed the way opening a tab/window has worked for as long as the internet and browsers have existed and foisted your precious little preference on every last one of your end users without so much as announcing it or providing an opt-out for people who just want it to work the way it's always worked?  

This has to be the single most arrogant, short-sighted, self-important decision by a software company since Apple decided they knew better than everyone else and removed the back button.  In a way it's a good thing because that single lost hour of my time has prevented me from making the mistake of investing further time or resources supporting this project. It does more than that though, it gives me a few new names (Zen principals who would have been aware of this change) to add to my list of people whose involvement in any project of any type is a huge red flag and probably a basis not to contribute in and of itself.  Just stunningly, breathtakingly cringe and customer-hostile behavior.  Like whoever's idea this was is just a low quality person in general and anyone who went along with it is a mediocre pushover at best; it's just incredibly childish and repulsive behavior by a developer in any context.

**Reply by mr-cheffy** [reply]
Sorry, but... what are you talking about?

**Reply by tobias-hay** [reply]
are you talking about opening new tabs beings a command bar style? In which case if so you can disable that.

**Reply by patillacode** [reply]
The biggest joke, besides your name, is coming here behind a fake profile to rant about something that is awesome and **free**.

Yeah, maybe it is not for everyone and of course you can disagree but you can also just give your opinion in a calm, educated way too. 

If you don't like it you can not use it, make your own, suggest changes, give ideas... but this... just a big fat rant is probably more childish and repulsive behaviour than what you are complaining about.

Congrats on your degree, btw. We all needed to know about it.

**Reply by DamienBlackwood** [reply]
What the fuck is this guy crying about??

**Comment by 74796a** [comment]
I won't be the only one leaving i can promise you that. you did something incredibly stupid, everyone told you how stupid it was, so now you're so triggered you're sending emails to argue with random users. *edited out profanity*

**Reply by pressRtowin** [reply]
> you're sending emails to argue with random users

Imagine being so thick that you don't understand how notifications on GitHub work. 

> probably a basis not to contribute in and of itself

You say this like you've made any contributions here.

> customer-hostile

You say this like you're paying to use it lmfao.

> from a UX design perspective which i have a degree in

God I feel sorry for anyone who has the misfortune of working with you.

Kindly shut the hell up. (someone please mute this guy)

**Reply by Propheticus** [reply]
![th](https://github.com/user-attachments/assets/c0ba075b-1f57-4272-a935-968322dd2402)

**Comment by 74796a** [comment]
yeah i'm not reading all that. revert quickly or you'll be irrelevant in a few weeks.

**Reply by pressRtowin** [reply]
tl;dr for your brain rot attention span: shut up and contribute something meaningful or shut up indefinitely

**Reply by pressRtowin** [reply]
Not to mention the hilarious point that the comment you quoted has literally nothing to do with you but you couldn't figure that out since you have no clue how github email notifications work.

**Comment by 74796a** [comment]
or i'll just come back and say the same thing under another acct in a few weeks since it apparently ruined everyone's day over there to be told the truth lmao, what a pack of losers

**Reply by pressRtowin** [reply]
🤡

Edit:

Oops my finger slipped
![Screenshot_20250415_181506_Gmail](https://github.com/user-attachments/assets/b0a24f57-22aa-4c8b-b7a7-82701b981dd9)

**Reply by whatthebassett** [reply]
I think I lost at least 5 to 10 IQ points reading this inane dribble. Go back to The Browser Company and suck on their teets, mmkay?

**Reply by mr-cheffy** [reply]
I need to save this one for top 2025 trolls, this is hilarious 

> or i'll just come back and say the same thing under another acct in a few weeks since it apparently ruined everyone's day over there to be told the truth lmao, what a pack of losers

Quite the contrary, thanks for the laughs!

**Comment by jd-santos** [comment]
I don't know what's happening in the thread up there, but any updates on direction for this now that tab groups has been released by Firefox?

**Reply by rijenkii** [reply]
https://github.com/zen-browser/desktop/pull/7359#issuecomment-2833459477

**Comment by shindi-renuo** [comment]
Can we rebase & merge https://github.com/zen-browser/desktop/pull/7359#issuecomment-2833459477 please?

**Reply by edurguti** [reply]
did they do anything here? The comments are locked

**Comment by mr-cheffy** [comment]
I'm locking this thread up as it's getting a bit... Interesting.

The pr #7359 contains folders and they are on they way 👍
